src/core/authenticator.cpp

```diff
@@ -0,0 +1,17 @@
+#include "authenticator.hpp"
+
+Authenticator::Authenticator()
+    : database(
```

Comment:
I suppose that it should rather get the database in constructor that create it itself